src/Support/InferExtensions/ModelExtension.php

```diff
@@ -72,22 +72,34 @@ public function getPropertyType(PropertyFetchEvent $event): ?Type
         throw new \LogicException('Should not happen');
     }
 
-    private function getAttributeTypeFromDbColumnType(string $columnType): AbstractType
+    /**
+     * MySQL/MariaDB decimal is mapped to a string by PDO.
+     * Floating point numbers and decimals are all mapped to strings when using the pgsql driver.
+     */
+    private function getAttributeTypeFromDbColumnType(?string $columnType, ?string $dbDriverName): AbstractType
     {
-        $type = Str::before($columnType, ' ');
-        $typeName = Str::before($type, '(');
-
-        // @todo Fix to native types
-        $attributeType = match ($typeName) {
-            'int', 'integer', 'bigint' => new IntegerType(),
-            'float', 'double', 'decimal' => new FloatType(),
-            'varchar', 'string', 'text', 'datetime' => new StringType(), // string, text - needed?
-            'tinyint', 'bool', 'boolean' => new BooleanType(), // bool, boolean - needed?
-            'json', 'array' => new ArrayType(),
-            default => new UnknownType("unimplemented DB column type [$type]"),
-        };
+        if ($columnType === null) {
+            return new UnknownType();
```

Comment:
Should we keep info about a column type being not implemented? Current implementation returns `StringType` if type is not handled which may hide debugging info in such cases.